ceph.go

```diff
@@ -605,6 +608,136 @@ func pgQuery(pgid string) *pgQueryOut {
 	return &out
 }
 
+func crushCmp(fp string) ([]pgMapping, error) {
+	out, err := runCrushCmp(fp)
+	if err != nil {
+		panic(err)
+	}
+
+	puis, err := parseCrushDiff(out)
+	if err != nil {
+		panic(err)
+	}
+
+	mappings := make([]pgMapping, 0, len(puis))
+	for _, pui := range puis {
+		for _, m := range pui.Mappings {
+			if m.From == m.To {
+				continue
+			}
+
+			mappings = append(mappings, pgMapping{
+				PgID: pui.PgID,
+				Mapping: mapping{
+					From: m.From,
+					To:   m.To,
+				},
+			})
+		}
+	}
+
+	return mappings, nil
+}
+
+func parseCrushDiff(in string) ([]*pgUpmapItem, error) {
+	var (
+		sc     = bufio.NewScanner(strings.NewReader(in))
+		puiMap = make(map[string]*pgUpmapItem)
+	)
+
+	for sc.Scan() {
+		line := strings.TrimSpace(sc.Text())
+
+		// For PGs that are part of a 3x replicated pool, each
```

Comment:
Maybe append ", for example" after "3x replicated pool"? I almost read this as this code being specific to 3xrepl pools.

Reply:
It was indeed confusing! Updated it.